server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -38,7 +41,7 @@ class OngoingSegmentReplications {
     private final RecoverySettings recoverySettings;
     private final IndicesService indicesService;
     private final Map<ReplicationCheckpoint, CopyState> copyStateMap;
-    private final Map<DiscoveryNode, SegmentReplicationSourceHandler> nodesToHandlers;
+    private final Map<String, SegmentReplicationSourceHandler> allocationIdToHandlers;
```

Comment:
Should we also remove the handler from `allocationIdToHandlers` after the segments have been copied?